episodes/03-muscle-connect.md

```diff
@@ -0,0 +1,418 @@
+---
+title: "MUSCLE3 - connecting a model to MUSCLE"
+teaching: 60
+exercises: 5
+---
+
+::: questions
+
+- How do you connect an existing python model code to MUSCLE3?
+
+:::
+
+::: objectives
+
+- Identify the inputs and outputs of your submodel and link them to ports
+- Recognize the Submodel Execution Loop structure in your code
+- Learn to connect your own model to the MUSCLE3 library
+
+:::
+
+## Introduction
+
+MUSCLE3 is the third incarnation of the Multiscale Coupling Library and Environment. Its purpose is to make creating coupled multiscale simulations easy, and to then enable efficient Uncertainty Quantification of such models using advanced semi-intrusive algorithms.
+
+MUSCLE3 uses the Multiscale Modelling and Simulation Language (MMSL) to describe the structure of a multiscale model and you will notice that the terminology in MUSCLE3 closely links to what you have learned in the previous episode about MMSF.
+
+MUSCLE3 consist of three components:
+
+**libmuscle** is a library that is used to connect component intances (e.g new or existing submodels) to a coupled simulation.
+
+**muscle_manager** is the central manager of MUSCLE3 that starts up submodels and coordinates their interaction.
+
+**ymmsl-python** is a Python library that contains class definitions to represent an MMSL model description. A yMMSL file serves as the interface between us (humans) and the muscle manager and is meant to describe the multiscale simulation and tell the muscle manager what to do.
+
+We continue with the reaction-diffusion model example that we used the previous episode and in this episode we will connect an implementation of the reaction model to the MUSCLE3 library step by step.
+
+## Disect your model
+
+In the previous episode we have discussed the Submodel Execution Loop (SEL) and the various operators that are associated with it.
+
+::: challenge
+
+## Exercise 1: Can you recognize the Submodel Execution Loop?
```

Comment:
It could first be helpful to the learners if we introduce the (admittedly simple) reaction model - what is it actually modelling? (Or will this be in the earlier theory episode already?)

Reply:
That depends a bit on Lourens' theory section. I suppose it would fit in there quite well since the model gets introduced there. But maybe a short description/walk through the code would fit here. I will add it.